### DISCUSSION TITLE
cpp/memory/understanding_of_pointers

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#25] [Announcements]
# cpp/memory/understanding_of_pointers

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/memory/understanding_of_pointers.html

**Comment by wOvOpy** [comment]
tql，我真的吹爆，看着是享受

**Comment by DuoAp** [comment]
大佬, 有个疑问. 假如我创建了个 Class A的指针 ptrA, 用ptrA指Class B的对象 应该会报错吧, 所以 C++的指针除了包含一个地址外, 是不是还有一套机制用来判断类型. 因为很多教程里大家都说指针就是保存地址的变量, 都不怎么说指针类型的事情.

**Reply by imarvinle** [reply]
解引用的时候，或者使用指针调用函数的时候会报错

**Reply by AndrewGuo0930** [reply]
这个类型信息可能不是存储在指针变量本身中的，而是由编译器进行管理和检查的？
